Removes Suicide Status

Incoming5643 opened the pull request:
You can still commit suicide, but even after committing suicide you can be revived.

Adds Quit Round/Do Not Resuscitate. This is a ghost verb you can use to tell people "I'm done with this round but I don't want to leave the server". This can be set regardless of how you died. Once you set this status most forms of revives will fail and the few that don't will just leave a catatonic body as if you live ghosted. Your corpse will appear as if your ghost didn't exist. Once you opt into this, you can't back out.

There's a million ways to kill yourself without suiciding, so this helps clear up confusion on what the dead's intentions are.

Incoming5643 committed:
Removes Suicide Status. You can still commit suicide, but even after committing suicide you can be revived.

Adds Quit Round/Do Not Resuscitate. This is a ghost verb you can use to tell people "I'm done with this round but I don't want to leave the server". This can be set regardless of how you died. Once you set this status most forms of revives will fail and the few that don't will just leave a catatonic body as if you live ghosted. Your corpse will appear as if your ghost didn't exist.

Incoming5643 commented:
(the reason the verb is called Quit Round is because if it was "Do Not Resuscitate" It would appear as the first verb for ghosts, which is annoying)

RemieRichards commented:
Your new var isn't set, so I assume WIP?

Incoming5643 commented:
No need for a new var, this runs on can_reenter_corpse

RemieRichards commented:
@Incoming5643 so remove your new var then poopy

RemieRichards commented on code/modules/mob/dead/observer/observer.dm:
```diff
@@ -17,6 +17,7 @@ var/list/image/ghost_darkness_images = list() //this is a list of images for thi
 	var/started_as_observer //This variable is set to 1 when you enter the game as an observer.
 							//If you died in the game and are a ghsot - this will remain as null.
 							//Note that this is not a reliable way to determine if admins started as observers, since they change mobs a lot.
+	var/do_not_resusitate = 0 //Player has a corpse but doesn't want to play, bodies will revive, but will end up souless
```
this one you butt

Incoming5643 replied:
oh whoops.

Incoming5643 committed:
cleans up a var I made but ended up not using

Shadowlight213 commented:
@Incoming5643 Can you set it so that the do not revive gets automatically set if they have a loyalty implant in? This might help discourage shitters who suicide during rev after getting implanted.

PKPenguin321 commented:
>Can you set it so that the do not revive gets automatically set if they have a loyalty implant in? This might help discourage shitters who suicide during rev after getting implanted.

that wont stop them from killing themselves and not re-entering their corpse
it wouldn't accomplish anything

optimumtact commented:
yeah please stop trying to punish or prevent people from removing themselves from a situation they don't want to be in.

Dorsisdwarf commented:
suiciding with an implant should vomit up the implant so sec didn't just waste their most important resource on your lazy ass

Bawhoppen commented:
OR it does remove the implant you shitter

Jordie0608 commented on code/game/gamemodes/game_mode.dm:
```diff
@@ -436,9 +436,6 @@
 				continue //AFK client
 			if(L.stat)
-				if(L.suiciding)	//Suicider
-					msg += "<b>[L.name]</b> ([L.ckey]), the [L.job] (<span class='boldannounce'>Suicide</span>)\n"
```
This is useful for admins, I'd be better if it wasn't removed.

Incoming5643 replied:
As far as I can tell that status only occurs when someone commits suicide but is still in their body.  This isn't really a normal state to be in. In this pull the guy who burned his hands 20 times on a light and the guy who blew up an airtank are treated equal until one of them says "yeah peace I'm out!" and uses the verb.

Cookiesndip commented:
---. Removing QoL for medbay, now medics will waste charge in defibs and time futilely trying to resurrect suicides. No reason for this change other than to make everything more unwieldy and to remove fun.

It also murkies the waters so you can't drag around suicided corpses and do things with them with people not being able to tell that it was a suicide, thus getting you lynched very quickly.

Incoming5643 commented:
We need it because sometimes people intentionally bash their heads in or space themselves then get found and it looks for all intents like a murder. So they bring it in and try to clone it, at which point the clonee either gets spammed with clone messages, or gets cloned, politely tells the geneticist they're wasting their time, and then kills themself again.

People who use this message get the "the spirit has departed" message. It's the equivalent of trying to heal the vacant; sure it's technically possible, but most doctors aren't going to prioritize it and many won't even give the attempt. As far as I'm concerned once a body is in a morgue tray it's free game for shenanigans. If someone gets bwoiked from a tray found body the suspicion should be on the person who put it there, not the person who took it.

If people need a more descriptive message that states definitively "THIS PERSON DOESN'T CARE ABOUT THIS CORPSE ANYMORE" we can do that, it's just nice to cut down on a variable.

Alek2ander commented:
Players should be able to see if someone has quit the round when examining the corpse.
Also, I think the status should be set for the mob corpse, not the ghost, so it can be applied if the ghost gets a new character (or the mob is unlinked from the mind in general, like when being borged)

Supermichael777 commented:
acualy there are situations where having a borged guys body is useful. see ling

Dorsisdwarf commented:
Solution: Have DNR, but *also* keep suicide status (Which just trips DNR permanently)

bgobandit commented:
^

+1

Incoming5643 committed:
Requested Changes:
*Committing suicide now automatically sets do not resuscitate
*Committing suicide now automatically ghosts afterwards

phil235 commented on code/modules/mob/dead/observer/observer.dm:
```diff
@@ -147,6 +147,18 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
+		return
+	if(alert(src, "Setting this option will keep you from reentering your body for the rest of the round. You cannot take this choice back!", "Do Not Resuscitate", "Proceed", "Nevermind!") == "Nevermind!")
+		return
+	can_reenter_corpse = 0
```
If people can't be bothered to commit suicide to be properly removed from the round, what makes you think players will remember there's a do no resuscitate that they can use? The verb is in the ghost tab and most people won't even realize it exists. The players also get nothing from using this verb, it's only there to help people still playing the round.

Bawhoppen commented:
Suicide verb should just be a faster way to actually die. Theres no reason killing yourself with a fire extinguisher would be different than someone else killing you.

Incoming5643 commented:
As is so often the case, extended examination of the milk has cause it to mentally sour for me.
